edb/schema/delta.py

```diff
@@ -1405,6 +1406,36 @@ def _cmd_tree_from_ast(cls, schema, astnode, context):
                 schema,
                 context.modaliases,
             )
+        elif field.name == 'required' and not new_value:
+            # disallow dropping required that is not locally set
+            parent_obj = parent_op.get_object(schema, context, default=None)
+            errmsg = None
+
+            if parent_obj:
+                local_val = parent_obj.get_explicit_local_field_value(
+                    schema, 'required', None)
+
+                if local_val is not True:
```

Comment:
`if not local_val`?

Please avoid using `is True` or `is False` checks in Python code.

Reply:
OK, I'll change it to truthy test. I'll also change the variable name to `local_required` so that the condition reads as `if not local_required:` which conveys the intent a little better, I think.